src/AdoNetCore.AseClient/AseDataReader.cs

```diff
@@ -9,7 +10,7 @@ namespace AdoNetCore.AseClient
 {
     public sealed class AseDataReader : DbDataReader
     {
-        private readonly TableResult[] _results;
+        private readonly IEnumerator<TableResult> _results;
```

Comment:
I had a crack at it, and the issue is that the reader exposes the number of rows in each `TableResult`. It's not clear what you put in there if you don't know. Maybe +1 as each row goes in. I backed it out in the end because I wasn't sure. It would be cool for a very large result set - and do-able if we can suss out what other providers do.

Reply:
Good point